### ISSUE TITLE
Why do the installation instructions encourage "sudo pip install"

### ISSUE THREAD

**Issue by robvdl** [motioneye-project/motioneye#3066]
First of all, I personally don't run this project, but I am a Python developer and a friend came across it and installed it.

I was a bit shocked to see "sudo pip install" instructions, especially because that is going away in the future.

https://stackoverflow.com/questions/61452582/why-is-using-sudo-pip-a-bad-idea

Why are we encouraging this sort of installation? Why can't we recommend users to create a virtualenv instead and teach good practices?

I don't want to come across negative, but "sudo pip install" is always a bad idea.

**Comment by MichaIng** [comment]
motionEye has a low number of dependencies without any specific version requirements. So IMO a venv/virtualenv does not add any benefit. If anyone is running into any conflicts with the system-wide installation, I would be interested to know, since it must be a very rare and special case.

However, you are right that Python generally moves towards venv only, recet Debian and Ubuntu install flag files with their distro packages which prevent `pip` from installing system-wide, without adding another flag or config option. So yes, we should switch to venv in our install instructions.

Please open a PR for the readme, if someone finds time. Also the install tests should be updated accordingly.

**Comment by robvdl** [comment]
I dunno if I agree with the "does not add any benefits" because my friend did run into conflicts. I'll have a look at it when I can, but I'm going to try zoneminder with that setup instead.

There has to be a better way to install it. But I thought that we got over that "sudo pip install" thing years ago, almost nobody does that anymore.

**Comment by MichaIng** [comment]
Did you even read the second half of my comment?

**Comment by robvdl** [comment]
You're taking about the `--user` flag?

**Comment by MichaIng** [comment]
No, that is still prevented on Debian and Ubuntu. This part of my comment is what I mean:

> So yes, we should switch to venv in our install instructions.
> 
> Please open a PR for the readme, if someone finds time. Also the install tests should be updated accordingly.

It is one more package `python3-venv` and two more commands to set it up and load it. The problem is that the systemd service needs to be changed as well. Not yet sure how to do this best. We would need to detect the venv in the `linux_init` script, if present, and in that case load it and get the `meyectl` path accordingly.

**Comment by robvdl** [comment]
I'll have to have a good read of the code first to get an understanding of the installation, it looks like motioneye_init.py invokes a shell script which makes me wonder what the point of that is and why not just run the shell script directly. But there may have been a reason.

Makes me wonder "who not just `make install`" when there already is a Makefile there in the project root.

**Comment by MichaIng** [comment]
Here is the shell script and systemd unit: https://github.com/motioneye-project/motioneye/tree/dev/motioneye/extra

I was wondering as well what the Python script purpose is. At least it is installed into PATH. However now it comes in handy, since it assures the script is executed within the venv, so that we can know the location of `meyectl`.

**Comment by MichaIng** [comment]
In DietPi, we currently do this:
```sh
python3 -m venv --upgrade-deps /opt/motioneye
/opt/motioneye/bin/pip3 install -U motioneye
/opt/motioneye/bin/motioneye_init
sed -i 's|ExecStart=.*/meyectl|ExecStart=/opt/motioneye/bin/meyectl|' /etc/systemd/system/motioneye.service
```
I am wondering whether there is a sane way to autodetect the correct `meyectl` path in https://github.com/motioneye-project/motioneye/blob/dev/motioneye/extra/linux_init:
- If in `PATH`, that should be fine. `/opt/motioneye/bin/activate` should achieve this, and `motioneye_init` should preserve it when calling `linux_init`.
- If not in `PATH`, maybe `motioneye_init` executable location can be obtained, and `meyectl` should be in the same directory. If difficult to obtain within `linux_init`, `motioneye_init` can pass it as argument. A new option for `linux_init` to specify the `meyectl` path makes sense anyway. That would then take precedence over autodetection.
- It could be derived from the `linux_init` path as well, if the environment follows standard structure:
  ```sh
  echo "$(realpath "$(dirpath "$BASH_SOURCE")/../../../bin/meyectl")"
  ```
